### PR TITLE
Improve IPv6 testing and minor fixes

### PR DESCRIPTION
**Pull request by aojea**
Aggregated several commits in the PR because I was finding several issues.

It was reported that the cluster-network-operator wasn't configuring correctly kube-proxy bind-address if it was "::", so I added some unit tests to check that it works correctly.

The other fixes are explained in the commit messages.

### CODE REVIEW

**Comment by rcarrillocruz** [comment]
you may want to rebase to single commit

**Comment by aojea** [comment]
> you may want to rebase to single commit

I prefer not to, since every commit is totally independent and, in case of an error, it will be easier to revert just the offending one.

File: pkg/network/kube_proxy_test.go [hunk 1]
```diff
@@ -98,6 +127,66 @@ winkernel:
...
+	errs := validateKubeProxy(&configIPv6)
+	g.Expect(errs).To(HaveLen(0))
+
+	cfg, err := kubeProxyConfiguration(map[string]operv1.ProxyArgumentList{
+		// special address+port combo
+		"metrics-bind-address":   {"fd00:1234::4"},
+		"metrics-port":           {"999"},
+		"conntrack-max-per-core": {"10"},
+	},
+		&configIPv6,
```
Comment (danwinship) [inline pkg/network/kube_proxy_test.go:142]: I don't like the way this indents; can you add a newline before `map` so the rest of the map body up to the `},` indents another level?

File: pkg/network/kube_proxy_test.go [hunk 2]
```diff
@@ -98,6 +127,66 @@ winkernel:
...
+func TestKubeProxyIPv6Config(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	errs := validateKubeProxy(&configIPv6)
+	g.Expect(errs).To(HaveLen(0))
+
+	cfg, err := kubeProxyConfiguration(map[string]operv1.ProxyArgumentList{
+		// special address+port combo
+		"metrics-bind-address":   {"fd00:1234::4"},
+		"metrics-port":           {"999"},
```
Comment (danwinship) [inline pkg/network/kube_proxy_test.go:139]: use a port above 32767 here to test that fix?

**Comment by danwinship** [comment]
> > you may want to rebase to single commit
> 
> I prefer not to, since every commit is totally independent and, in case of an error, it will be easier to revert just the offending one.

yeah. We don't want PRs with one real commit and a bunch of fixes to it, but if the commit is logically made up of multiple pieces, go with multiple commits

File: pkg/network/kube_proxy_test.go (hunk 1 above)
Comment (aojea) [reply pkg/network/kube_proxy_test.go:143]: done,

File: pkg/network/kube_proxy_test.go (hunk 2 above)
Comment (aojea) [reply pkg/network/kube_proxy_test.go]: done

**Pull request merged**